ayanda/src/main/java/sintulabs/p2p/Lan.java

```
@@ -0,0 +1,330 @@
+import android.widget.Toast;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by sabzo on 12/26/17.
+ */
+
+public class Lan extends P2P{
+    // constants for identifying service and service type
+    public final static String SERVICE_NAME_DEFAULT = "NSDaya";
+    public final static String SERVICE_TYPE = "_http._tcp.";
+    public final static String LAN_DEVICE_NUM_UPDATE = "AYANDA_LAN_DEVICE_UPDATE";
+    public final static String LAN_SERVICE_LOST = "AYANDA_LAN_SERVICE_LOST";
+    // For discovery
+    private NsdManager.DiscoveryListener mDiscoveryListener;
+    private NsdManager.ResolveListener mResolveListener;
+    // For announcing service
+    private int mLocalPort;
+    private Context mContext;
+    private String mServiceName;
+    private NsdManager.RegistrationListener mRegistrationListener;
+
+    private NsdManager mNsdManager;
+    private Boolean serviceAnnounced;
+
+    private List<Device> deviceList;
+
+    private Set<String> servicesDiscovered;
+
+    private ILan iLan;
+
+    public Lan(Context context, ILan iLan) {
+        mContext = context;
+        this.iLan = iLan;
+        mNsdManager = (NsdManager) context.getSystemService(Context.NSD_SERVICE);
+        deviceList = new ArrayList<>();
+        serviceAnnounced = false;
+        servicesDiscovered = new HashSet<>();
+    }
+
+    @Override
+    public Boolean isSupported() {
+        return null;
+    }
+
+    @Override
+    public Boolean isEnabled() {
+        return null;
+    }
+
+    @Override
+    public void announce() {
+        // Create the NsdServiceInfo object, and populate it.
+        NsdServiceInfo serviceInfo  = new NsdServiceInfo();
+        int port = 0;
+        try {
+            port = findOpenSocket();
+        } catch (IOException e) {
+            e.printStackTrace();        }
```

should probably throw an exception?